presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestAnalyzer.java

```diff
@@ -1007,6 +1008,16 @@ public void testInValidJoinOnClause()
         assertFails(TYPE_MISMATCH, "SELECT * FROM (VALUES (2, 2)) a(x,y) JOIN (VALUES (2, 2)) b(x,y) ON (a.x=b.x, a.y=b.y)");
     }
 
+    @Test
+    public void testQuantifiedComparisonExpression()
+            throws Exception
+    {
+        analyze("SELECT * FROM t1 WHERE t1.a <= ALL (VALUES 10, 20)");
+        assertFails(MULTIPLE_FIELDS_FROM_SUBQUERY, "SELECT * FROM t1 WHERE t1.a = ANY (SELECT 1, 2)");
+        assertFails(TYPE_MISMATCH, "SELECT * FROM t1 WHERE t1.a = SOME (VALUES ('abc'))");
+        assertFails(MISMATCHED_SET_COLUMN_TYPES, "SELECT * FROM t1 WHERE t1.a = ALL (VALUES ROW(10), ROW('abc'))");
```

Comment:
ROW constructors are unnecessary. You can just do `VALUES 10, 'abc'`.

BTW, this might not be testing the type analysis for quantified comparison at all -- it may be failing in the analysis of VALUES.